src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2821,18 +2852,19 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
 
             if ((_compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X86
                 || _compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X64)
-#if READYTORUN
-                && isMethodDefinedInCoreLib()
-#endif
                )
             {
 #if !READYTORUN
                 // This list needs to match the list of intrinsics we can generate detection code for
                 // in HardwareIntrinsicHelpers.EmitIsSupportedIL.
 #else
-                // For ReadyToRun, this list needs to match up with the behavior of FilterNamedIntrinsicMethodAttribs
-                // In particular, that this list of supported hardware will not generate non-SSE2 safe instruction
-                // sequences when paired with the behavior in FilterNamedIntrinsicMethodAttribs
+                // For ReadyToRun we set these hardware features as enabled always, as the overwhelming majority
```

Comment:
For reference, this is what the x86 emulator for Windows 10 on ARM reports (for both my Samsung GalaxyBook 2 and my Microsoft Surface Pro X):
![Capture2](https://user-images.githubusercontent.com/10487869/76110927-3a87d380-5f94-11ea-8488-c2df98401db3.PNG)

Notably, SSE4.2, POPCNT, and LZCNT aren't supported.
* SSE4.2 and POPCNT were introduced in November 2008, so 11 years old
* LZCNT is newer than both and is June 2013, so 6 years old
  * AMD introduced LZCNT and POPCNT in 2007 (this was part of their ABM instruction set and uses the same CPUID bits, respectively, that Intel later used)
  * LZCNT, for Intel, is actually newer than AVX (2011) and same time frame as AVX2 (2013)
  * BMI1/BMI2 are the same time frame as AVX2 (2013, Haswell)

Reply:
Oh darn, I had thought that SSE4.2 was supported on the emulator, the general strategy here is optimized based on supporting a set that is extremely widely available. As the Arm emulation system only supports SSE4.1, what are your thoughts around changing the SIMDSupportLevel of SIMD_SSE4_Supported to only require SSE4.1? As far as I can tell, the only difference that the jit takes is around support for the pcmpgtq instruction. If we change line 215 in simdcodegenxarch.cpp to check for compSupports(Sse42) instead of using getSIMDSupportLevel() >= SIMD_SSE4_Supported, and changing getSIMDSupportLevel to only require Sse41, I think that would be the only change needed.

Reply:
One detail is that supporting a few extra hardware instruction sets isn't problematic, and causes only minimal jit behavior at runtime, but the difference between Sse41 and Sse42 support is very significant as the jit has the concept of simdsupport level which is quite impactful.

Reply:
I think that would be fine, the biggest improvements are likely to come from SSE4.1 anyways, but I would appreciate hearing @CarolEidt's thoughts as well.

Reply:
Honestly, I don't feel strongly about this, but it seems reasonable to me. One wonders if there are any plans to update the emulator.

Reply:
I find it interesting that QEMU's CPU emulator has exactly the same restriction to only supporting SSE4.1. I don't intend to put the change into this PR (as it is too large already), but will prepare something separate when this PR is complete.